### ISSUE TITLE
Vignette Error: differing number of rows 

### ISSUE THREAD

**Issue by juliaszusz** [pcahan1/singleCellNet#25]
Hello,

I am trying to work through the vignette using the example data provided, and am running into an error during the "Classification Annotation assignment" step:

**stPark <- get_cate(classRes = crParkall, sampTab = stPark, dLevel = "description1", sid = "sample_name", nrand = nqRand)**

**Error in data.frame(..., check.names = FALSE) : arguments imply differing number of rows: 43745, 43795**

5. stop(gettextf("arguments imply differing number of rows: %s", paste(unique(nrows), collapse = ", ")), domain = NA)
4. data.frame(..., check.names = FALSE)
3. cbind(deparse.level, ...)
2. cbind(sampTab, category = topCats, scn_score = topCat_score)
1. get_cate(classRes = crParkall, sampTab = stPark, dLevel = "description1", sid = "sample_name", nrand = nqRand)

> sessionInfo()
R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.2

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_CA.UTF-8/en_CA.UTF-8/en_CA.UTF-8/C/en_CA.UTF-8/en_CA.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] singleCellNet_0.1.0 cowplot_1.0.0       reshape2_1.4.4     
[4] pheatmap_1.0.12     dplyr_0.8.5         ggplot2_3.3.0      

loaded via a namespace (and not attached):
  [1] backports_1.1.6     sn_1.6-1            plyr_1.8.6         
  [4] igraph_1.2.5        lazyeval_0.2.2      splines_3.6.2      
  [7] listenv_0.8.0       usethis_1.6.0       TH.data_1.0-10     
 [10] digest_0.6.25       htmltools_0.4.0     gdata_2.18.0       
 [13] fansi_0.4.1         magrittr_1.5        memoise_1.1.0      
 [16] cluster_2.1.0       ROCR_1.0-7          remotes_2.1.1      
 [19] globals_0.12.5      sandwich_2.5-1      prettyunits_1.1.1  
 [22] colorspace_1.4-1    ggrepel_0.8.2       xfun_0.13          
 [25] callr_3.4.3         crayon_1.3.4        jsonlite_1.6.1     
 [28] survival_3.1-12     zoo_1.8-7           ape_5.3            
 [31] glue_1.4.0          gtable_0.3.0        leiden_0.3.3       
 [34] pkgbuild_1.0.6      future.apply_1.4.0  BiocGenerics_0.30.0
 [37] scales_1.1.0        mvtnorm_1.1-0       bibtex_0.4.2.2     
 [40] Rcpp_1.0.4          metap_1.3           plotrix_3.7-7      
 [43] viridisLite_0.3.0   reticulate_1.15     rsvd_1.0.3         
 [46] stats4_3.6.2        htmlwidgets_1.5.1   httr_1.4.1         
 [49] gplots_3.0.3        RColorBrewer_1.1-2  TFisher_0.2.0      
 [52] ellipsis_0.3.0      ica_1.0-2           pkgconfig_2.0.3    
 [55] farver_2.0.3        tidyselect_1.0.0    labeling_0.3       
 [58] rlang_0.4.5         munsell_0.5.0       tools_3.6.2        
 [61] cli_2.0.2           devtools_2.3.0      ggridges_0.5.2     
 [64] stringr_1.4.0       yaml_2.2.1          npsurv_0.4-0       
 [67] processx_3.4.2      knitr_1.28          fs_1.4.1           
 [70] fitdistrplus_1.0-14 caTools_1.18.0      purrr_0.3.3        
 [73] randomForest_4.6-14 RANN_2.6.1          pbapply_1.4-2      
 [76] future_1.16.0       nlme_3.1-147        compiler_3.6.2     
 [79] rstudioapi_0.11     plotly_4.9.2.1      curl_4.3           
 [82] png_0.1-7           testthat_2.3.2      lsei_1.2-0         
 [85] tibble_3.0.0        DescTools_0.99.34   stringi_1.4.6      
 [88] ps_1.3.2            desc_1.2.0          lattice_0.20-41    
 [91] Matrix_1.2-18       multtest_2.40.0     vctrs_0.2.4        
 [94] mutoss_0.1-12       pillar_1.4.3        lifecycle_0.2.0    
 [97] Rdpack_0.11-1       lmtest_0.9-37       RcppAnnoy_0.0.16   
[100] data.table_1.12.8   bitops_1.0-6        irlba_2.3.3        
[103] gbRd_0.4-11         patchwork_1.0.0     R6_2.4.1           
[106] KernSmooth_2.23-16  sessioninfo_1.1.1   codetools_0.2-16   
[109] boot_1.3-24         MASS_7.3-51.5       gtools_3.8.2       
[112] assertthat_0.2.1    pkgload_1.0.2       rprojroot_1.3-2    
[115] withr_2.1.2         mnormt_1.5-6        multcomp_1.4-13    
[118] expm_0.999-4        parallel_3.6.2      grid_3.6.2         
[121] tidyr_1.0.2         Rtsne_0.15          numDeriv_2016.8-1.1
[124] Biobase_2.44.0     

Any assistance would be greatly appreciated!
Thanks,

Julia

**Comment by yuqiyuqitan** [comment]
Hi Julia,

Thanks for reporting the bug and using singleCellNet. I identified and fixed the error. If you re-install singleCellNet from Github again, the error should go away.

Best,
Yuqi

**Comment by juliaszusz** [comment]
Hi Yuqi,

Thank you for responding so quickly. I reinstalled singleCellNet from github and went through again, but I am still having the same error come up.

Thanks,

Julia

**Comment by juliaszusz** [comment]
Thanks, it worked now and I made it through the rest of the vignette. I did find a minor thing a few lines later though:

**skylineClass(crParkall, "T cell", stKid2, "description1",.25, "sample_name")**
Error in scale_fill_viridis(discrete = TRUE) : 
  could not find function "scale_fill_viridis"

I then loaded the "viridis" package and it worked, but could that be easily added to the dependencies that load when loading your package?

I also had to load "umap" separately a littler later on for the "plot_umap" function.

Thanks again!

Julia
